### PR TITLE
remove usage of SERVING_LOAD_MODELS in examples/docs/tests

### PR DESCRIPTION
**Pull request by siddvenk**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

**Review by sindhuvahinis — APPROVED** [review]

File: engines/python/setup/djl_python/tests/test_test_model.py [hunk 1]
```diff
@@ -53,7 +53,6 @@ def test_all_code(self):
     def test_with_env(self):
         envs = {
             "OPTION_MODEL_ID": "NousResearch/Nous-Hermes-Llama2-13b",
```
Comment (frankfliu) [inline engines/python/setup/djl_python/tests/test_test_model.py:55]: HF_MODEL_ID

File: engines/python/setup/djl_python/tests/test_test_model.py [hunk 2]
```diff
@@ -110,7 +109,6 @@ def test_all_code_chat(self):
     def test_with_env_chat(self):
         envs = {
             "OPTION_MODEL_ID": "TheBloke/Llama-2-7B-Chat-fp16",
```
Comment (frankfliu) [inline engines/python/setup/djl_python/tests/test_test_model.py:111]: HF_MODEL_ID

File: serving/docs/lmi/deployment_guide/configurations.md [hunk 3]
```diff
@@ -125,18 +125,17 @@ You can find these configurations in the respective [user guides](../user_guides
 
 All LMI Configuration keys available in the `serving.properties` format can be specified as environment variables.
 
-The translation for `engine` is unique. The configuration `engine=<engine>` is translated to `SERVING_LOAD_MODELS=test::<engine>=/opt/ml/model`.
-For example:
-
-* `engine=Python` is translated to environment variable `SERVING_LOAD_MODELS=test::Python=/opt/ml/model`
```
Comment (frankfliu) [inline serving/docs/lmi/deployment_guide/configurations.md:131]: By default, LMI will use `engine=Python` will be used, we can use `OPTION_ENGINE=Python` to explictly set the engine.

You can also force use `OPTION_MPI_MODE=true` to force use MPI mode

File: engines/python/setup/djl_python/tests/test_test_model.py (hunk 1 above)
Comment (lanking520) [reply engines/python/setup/djl_python/tests/test_test_model.py:55]: HF_MODEL_ID actually clash with OPTION_MODEL_ID. Some configuration doesn't really work

File: engines/python/setup/djl_python/tests/test_test_model.py (hunk 2 above)
Comment (frankfliu) [reply engines/python/setup/djl_python/tests/test_test_model.py:111]: We should never use `OPTION_MODEL_ID`, I cannot think of a real use case that requires OPTION_MODEL_ID

File: engines/python/setup/djl_python/tests/test_test_model.py (hunk 1 above)
Comment (siddvenk) [reply engines/python/setup/djl_python/tests/test_test_model.py:55]: Yea ideally should be HF_MODEL_ID, but as Qing said some configurations don't really work with OPTION_MODEL_ID. I'll need to fix some parts of the test here, but will update to HF_MODEL_ID

Comment (frankfliu) [reply engines/python/setup/djl_python/tests/test_test_model.py:55]: In which case HF_MODEL_ID will clash with OPTION_MODEL_ID?

OPTION_MODEL_ID requires a model directory being defined already, while HF_MODEL_ID is treated as a new model.

**Comment by siddvenk** [comment]
I've included the removal of OPTION_MODEL_ID usage in this PR now.

File: serving/docs/lmi/deployment_guide/configurations.md [hunk 4]
```diff
@@ -125,17 +125,20 @@ You can find these configurations in the respective [user guides](../user_guides
...
 
-The translation for `engine` is unique. The configuration `engine=<engine>` is translated to `SERVING_LOAD_MODELS=test::<engine>=/opt/ml/model`.
-For example:
+The property `option.model_id` is unique. It is translated to `HF_MODEL_ID`.
 
-* `engine=Python` is translated to environment variable `SERVING_LOAD_MODELS=test::Python=/opt/ml/model`
-* `engine=MPI` is translated to environment variable `SERVING_LOAD_MODELS=test::MPI=/opt/ml/model`
+The property `engine` is translated to `OPTION_ENGINE`.
+By default, LMI will use the Python engine. You can use `OPTION_ENGINE=Python` to explicitly set the engine.
+To use the MPI engine, you should also provide `OPTION_MPI_MODE=true`.
```
Comment (frankfliu) [inline serving/docs/lmi/deployment_guide/configurations.md:132]: Avoid use "MPI engine" here,

If your engine requires use mpi launcher, you should also provide ...

File: serving/docs/lmi/deployment_guide/configurations.md [hunk 5]
```diff
@@ -125,17 +125,20 @@ You can find these configurations in the respective [user guides](../user_guides
 
 All LMI Configuration keys available in the `serving.properties` format can be specified as environment variables.
 
-The translation for `engine` is unique. The configuration `engine=<engine>` is translated to `SERVING_LOAD_MODELS=test::<engine>=/opt/ml/model`.
-For example:
+The property `option.model_id` is unique. It is translated to `HF_MODEL_ID`.
```
Comment (frankfliu) [inline serving/docs/lmi/deployment_guide/configurations.md:128]: option_model_id and HF_MODEL_ID  are different thing, we might want to document HF_MODEL_ID separately

**Comment by frankfliu** [comment]
Please rebase latest master branch

**Review by frankfliu — APPROVED** [review]

**Pull request merged**